### PR TITLE
Adding example for Dynamic Ceph

### PR DESCRIPTION
**Pull request by this-tech**
This is a full example for Dynamic Ceph.

### CODE REVIEW

File: install_config/storage_examples/ceph_rbd_dynamic_example.adoc [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+== Create pool for dynamic volumes
+
+It is recommended to create a pool for your dynamic volumes to live.
+
+[NOTE]
+====
+Using the default pool of rbd is an option, but not recommended. 
+====
+
+From a Ceph admin or Mon node, create a new pool for dynamic volumes
```
Comment (ahardin-rh) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: s/admin/administrator


    *MON* node

Comment (ahardin-rh) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: In general, I  see usage of *Mon* and *MON*. Is one more correct than the other? Not familiar with the term. We can just pick a way and keep consistent across the page

File: install_config/storage_examples/ceph_rbd_dynamic_example.adoc [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+----
+<1> This base64 key is generated on one of the Ceph MON nodes using the `ceph auth get-key client.admin | base64` command, then copying the output and pasting it as the secret key's value.
+<2> This is required for Ceph RBD to work with dynamic provisioning.
+====
+
+Save the secret definition to a file, for example *_ceph-secret.yaml_*,
+then create the secret:
+
+====
```
Comment (ahardin-rh) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: I think you can get rid of the ====s around the code blocks and just keep the ----s. We are moving away from using those in code blocks

File: install_config/storage_examples/ceph_rbd_dynamic_example.adoc [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+metadata:
+  name: ceph-user-secret
+  namespace: default
+data:
+  key: QVFCbEV4OVpmaGJtQ0JBQW55d2Z0NHZtcS96cE42SW1JVUQvekE9PQ== <1>
+type:
+  kubernetes.io/rbd <2>
+
+----
+<1> This base64 key is generated on one of the Ceph MON nodes using the `ceph auth get-key client.kube | base64` command, then copying the output and pasting it as the secret key's value.
```
Comment (ahardin-rh) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: *MON*

File: install_config/storage_examples/ceph_rbd_dynamic_example.adoc [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+<2> This is required for Ceph RBD to work with dynamic provisioning.
+====
+
+Save the secret definition to a file, for example *_ceph-user-secret.yaml_*,
```
Comment (ahardin-rh) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: If this part is a procedure, you can format this as numbered steps. For example, you can do:

    .Save the secret definition to a file...secret:
    +
    ----
    $ oc create -f ceph-user-secret.yaml
    secret "ceph-user-secret" created
    ----

The + will keep the bits below it inline with the step

File: install_config/storage_examples/ceph_rbd_dynamic_example.adoc [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+then create the secret:
+
+====
+----
+$ oc create -f ceph-user-secret.yaml
+secret "ceph-user-secret" created
+----
+====
+
```
Comment (ahardin-rh) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: Same here:

    .Verify that the secret was created:
    +
    ----
    # oc get secret ceph-user-secret
    NAME               TYPE                DATA      AGE
    ceph-user-secret   kubernetes.io/rbd   1         5d
    ----

File: install_config/storage_examples/ceph_rbd_dynamic_example.adoc [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+  kind: Secret
+  metadata:
+    name: ceph-user-secret
+  data:
+    key: QVFCbEV4OVpmaGJtQ0JBQW55d2Z0NHZtcS96cE42SW1JVUQvekE9PQ== <1>
+  type:
+    kubernetes.io/rbd
+...
+----
+<1> Place the key from ceph-user-secret here in base64 format. See xref:../install_config/storage_examples/ceph_example.adoc#using-ceph-rbd-creating-the-ceph-secret[Creating the Ceph Secret].
```
Comment (ahardin-rh) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: `ceph-user-secret`

File: install_config/storage_examples/ceph_rbd_dynamic_example.adoc [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+# oc create -f ceph-claim.yaml
+persistentvolumeclaim "ceph-claim" created
+
+#and verify the PVC was created and bound to the expected PV:
+# oc get pvc
+NAME         STATUS    VOLUME                                     CAPACITY   ACCESSMODES   AGE
+ceph-claim   Bound     pvc-f548d663-3cac-11e7-9937-0024e8650c7a   2Gi        RWO           1m
+                                 <1>
+----
+<1> the claim dynamically created a ceph rbd PV.
```
Comment (ahardin-rh) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: s/ceph rbd PV/Ceph RBD PV

**Comment by ahardin-rh** [comment]
@donnydavis Just some minor comments from me. Is this targeted for OCP 3.6?

**Comment by donnydavis** [comment]
I can confirm this is functional origin 1.5, if needed I can confirm on ocp 3.5

File: install_config/storage_examples/ceph_rbd_dynamic_example.adoc (hunk 1 above)
Comment (donnydavis) [inline install_config/storage_examples/ceph_rbd_dynamic_example.adoc]: Probably stick with the CAPS version of mon

**Comment by ahardin-rh** [comment]
@donnydavis Okay, sounds like it should be marked for OCP 3.5 then.

**Comment by ahardin-rh** [comment]
We also need to update the _topic_map.yml file to get the topic added

**Comment by ahardin-rh** [comment]
@donnydavis Did you want me to just take this over so that it gets moved along and added into the next release?

**Comment by donnydavis** [comment]
Yea, I kind dropped the ball on this one. I can confirm this doc works on origin 1.5. I am fine with you taking it over. Thanks @ahardin-rh

**Comment by ahardin-rh** [comment]
Closing in favor of https://github.com/openshift/openshift-docs/pull/4849

**Comment by ahardin-rh** [comment]
No worries! Thanks @donnydavis

**Pull request closed without merging**